test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -355,6 +356,16 @@ TEST(QuicPlatformTest, QuicCertUtils) {
   OPENSSL_free(static_cast<void*>(der));
 }
 
+TEST(QuicPlatformTest, QuicTestOutput) {
+  QuicLogThresholdSaver saver;
+
+  // Set log level to INFO to see the test output path in log.
+  quic::GetLogger().set_level(quic::INFO);
+
+  quic::QuicRecordTestOutput("quic_test_output.1", "output 1 content\n");
+  quic::QuicRecordTestOutput("quic_test_output.2", "output 2 content\n");
```

Comment:
This just tests the fast-fail due to QUIC_TEST_OUTPUT_DIR not being set, right?  It'd be nice to test that the filesystem work actually works and the various appends do as well.

Reply:
Agreed. If we end up using Api::fileSystem(), the test could inject a mock Filesystem::Instance by way of that, which would simplify verifying this.

Reply:
I tested it with --action_env=QUIC_TEST_OUTPUT_DIR=/tmp in the bazel command. (PR description has the full command).

Alternatively, we can do a setenv() at the beginning of the test, and check for the actual file content at the end of the test, which will make this test less manual. WDYT?

Reply:
Yeah, I think having CI actually cover the code path would be good.  I think setenv is probably the best bet but I'd be fine if there's a way to build it into the test definition too.

Reply:
I've updated the test to setenv(), if the env is not already set, so the code is now covered by CI.
Checking the output is still manual though, because the test does not know the file path, which is generated by the QuicRecordTestOutput function internally.

Reply:
Could we use the Envoy log utils (EXPECT_LOG_NOT_CONTAINS) to verify there was no warning logged, to make sure we didn't hit the fast fail paths?

I'd think we could verify existence of /tmp/quic_test_output.1 unless we expect that to change under the hood

Reply:
> Could we use the Envoy log utils (EXPECT_LOG_NOT_CONTAINS) to verify there was no warning logged, to make sure we didn't hit the fast fail paths?
> 
Good idea, done.

> I'd think we could verify existence of /tmp/quic_test_output.1 unless we expect that to change under the hood
The filename is not visible to the test, but I added EXPECT_LOG_CONTAINS, which make sure _some_ test output file is generated.